### ISSUE TITLE
crashes as soon as I interact with the screen after presenting master controller

### ISSUE THREAD

**Issue by techyowl** [ariok/BWWalkthrough#90]
I cannot seem to figure out why I'm getting a (lldb) crash with no thread details when I tap or swipe anywhere on the screen when presented from a view controller that is inside a navigation bar controller which is in a tabbarcontroller I have tried presenting it from the self.present as well as self.tabbarcontroller.present. it fails after you touch it. its weird because it renders the control you just cannot interact with it without crashing the whole app. the page controller updates correctly but you cannot use it since it crashes on any interaction. I have followed the tutorial listed in the read me I have also mimicked the example it keeps crashing. I'm using the cocoa pods build and I have tried it with out ui elements on the master controller as well it crashes just the same. Please help as I'd really like to you this pod in our app. Thank You.

**Comment by techyowl** [comment]
I have tried to use it by manually including the files and I have also looked through the other issues and cannot figure out why it keeps crashing. I have tried using the app delegates root view controller and the navigation controller to present the view it fails just the same. I have also tried it on a separate storyboard and I have tried disabling user interaction on the page view controllers. I have tried various castings and I have tried to disable the extra elements. I have tried everything I can think of. Im using swift 3.0 if that matters.

**Comment by techyowl** [comment]
Current Page 0
2017-02-24 17:44:03.990285 Cache[5362:1675767] -[MTLDebugResource opacity]: unrecognized selector sent to instance 0x174221d00

this is the error response that I'm getting.

2017-02-24 17:44:03.994891 Cache[5362:1675767] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '-[MTLDebugResource opacity]: unrecognized selector sent to instance 0x174221d00'
*** First throw call stack:
(0x1934dd1b8 0x191f1455c 0x1934e4268 0x1934e1270 0x1933da80c 0x1993ccf94 0x1993ccbbc 0x199b78d30 0x199b7be78 0x199b76130 0x19348ab5c 0x19348a4a4 0x1934880a4 0x1933b62b8 0x194e6a198 0x1993fd7fc 0x1993f8534 0x10023d648 0x1923995b8)
libc++abi.dylib: terminating with uncaught exception of type NSException
(lldb)

**Comment by techyowl** [comment]
this is how I am presenting it.

`
    func opentips(){
    
    // Get view controllers and build the walkthrough
    let stb = UIStoryboard(name: "WalkThroughs", bundle: nil)
    let walkthrough = stb.instantiateViewController(withIdentifier: "walk") as! BWWalkthroughViewController
    let page_zero = stb.instantiateViewController(withIdentifier: "wt00") as! BWWalkthroughPageViewController
    let page_one = stb.instantiateViewController(withIdentifier: "wt01") as! BWWalkthroughPageViewController
    
    // Attach the pages to the master
    walkthrough.delegate = self
    walkthrough.add(viewController:page_zero)
        
        walkthrough.add(viewController:page_one)
        //let appDelegate = UIApplication.shared.delegate as! AppDelegate
        self.present(walkthrough, animated: true, completion: nil)
        
}


// MARK: - Walkthrough delegate -

func walkthroughPageDidChange(_ pageNumber: Int) {
    print("Current Page \(pageNumber)")
}

func walkthroughCloseButtonPressed() {
    self.dismiss(animated: true, completion: nil)
}`

**Comment by techyowl** [comment]
I figured it out took me all day but here is what the issue was. I think because I'm presenting the BWWalkThroughViewController from a nested ViewController that is quite complex with auto layout. I needed to change the storyboard with the BWWalkThroughViewController so that its presentation was set to over fullscreen as opposed to the default value in the attributes inspector. setting this allows it to operate without crashing. I ended up using this as opposed to my own custom view controller I started building since I couldn't get this to work at the time. anyhow glad I was able to figure this silly bug out on my own sorry to bother you.
